source/extensions/quic_listeners/quiche/quic_filter_manager_connection_impl.cc

```diff
@@ -0,0 +1,130 @@
+#include "extensions/quic_listeners/quiche/quic_filter_manager_connection_impl.h"
+
+#include <memory>
+
+namespace Envoy {
+namespace Quic {
+
+QuicFilterManagerConnectionImpl::QuicFilterManagerConnectionImpl(
+    std::unique_ptr<EnvoyQuicConnection> connection, Event::Dispatcher& dispatcher)
+    : quic_connection_(std::move(connection)), filter_manager_(*this), dispatcher_(dispatcher),
+      stream_info_(dispatcher.timeSource()) {
+  // TODO(danzh): Use QUIC specific enum value.
+  stream_info_.protocol(Http::Protocol::Http2);
+}
+
+void QuicFilterManagerConnectionImpl::addWriteFilter(Network::WriteFilterSharedPtr filter) {
```

Comment:
What are we doing with filters here? I thought the idea is that we are going to skip all filters and just direct inject into an HCM codec? I'm probably just missing some context.

Reply:
Right I understand, but I don't think we actually need any network filters at all, right? Or you are relying on the standard filter manager to install the HCM as a filter essentially? I'm mainly trying to understand how much code we can cut out here, as I think everything is basically hard coded, right? Meaning, can't we just hard code install the HCM and basically skip all filter chain code?

Reply:
I don't think we can skip the filter chain code as HCM config need to be mapped to each connection anyway. If we don't use filter chain mechanism, we need to invent our own way to store up HCM config and do the mapping.